### PR TITLE
RdkitConformerFeaturizer

### PR DESCRIPTION
**Pull request by tonydavis629**
## Description

This PR implements a featurizer which uses RDkit to generate conformers and 3d coordinates

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
- [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
- [x] Run `mypy -p deepchem` and check no errors
- [x] Run `flake8 <modified file> --count` and check no errors
- [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/feat/molecule_featurizers/conformer_featurizer.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
+import numpy as np
+from rdkit import Chem
+from rdkit.Chem import AllChem
+from deepchem.feat.graph_data import GraphData
+from deepchem.feat import MolecularFeaturizer
+
+# similar to snap featurizer. both taken from Open Graph Benchmark (OGB) github.com/snap-stanford/ogb
```
Comment (rbharath) [inline deepchem/feat/molecule_featurizers/conformer_featurizer.py:7]: Could we add a couple comments on the differences from the snap list?

File: deepchem/feat/molecule_featurizers/conformer_featurizer.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        [
+            allowable_features['possible_bond_type_list'],
+            allowable_features['possible_bond_stereo_list'],
+            allowable_features['possible_is_conjugated_list']
+        ]))
+
+
+def safe_index(feature_list, e):
+    """
+    Return index of element e in list l. If e is not present, return the last index
```
Comment (rbharath) [inline deepchem/feat/molecule_featurizers/conformer_featurizer.py:66]: Can we add docstring (params/returns)

File: deepchem/feat/molecule_featurizers/conformer_featurizer.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    """
+    try:
+        return feature_list.index(e)
+    except ValueError:
+        return len(feature_list) - 1
+
+
+class RDKitConformerFeaturizer(MolecularFeaturizer):
+    """
+    A featurizer that featurizes an RDKit mol object as a GraphData object with 3D coordinates.
```
Comment (rbharath) [inline deepchem/feat/molecule_featurizers/conformer_featurizer.py:76]: Can you add more details about the conformer generation algorithm and a link to the RDKit docs?

File: deepchem/feat/molecule_featurizers/conformer_featurizer.py (hunk 2 above)
Comment (tonydavis629) [reply deepchem/feat/molecule_featurizers/conformer_featurizer.py:66]: done

File: deepchem/feat/molecule_featurizers/conformer_featurizer.py (hunk 3 above)
Comment (tonydavis629) [reply deepchem/feat/molecule_featurizers/conformer_featurizer.py:76]: done

File: deepchem/feat/tests/test_conformer_featurizer.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+def test_conformer_featurizer():
+    from deepchem.feat.molecule_featurizers.conformer_featurizer import RDKitConformerFeaturizer
+    smiles = ["C1=CC=NC=C1", "CC(=O)C", "C"]
+    featurizzer = RDKitConformerFeaturizer()
```
[the remark this answers is not in the document]
Comment (tonydavis629) [reply deepchem/feat/tests/test_conformer_featurizer.py:4]: fixed

**Comment by tonydavis629** [comment]
The main issue which I think we need to discuss is the handling of conformers in the featurization. We input one smiles, and get get out `n` number of conformers. How are those packaged? I think what makes the most sense is a list of GraphData objects. This requires a little bit different handling when doing batching, this is handled by flattening the batch and then forming a BatchGraphData, as seen in the doctest and unit test. The other alternative is a BatchGraphData object, but batching multiple BatchGraphData objects will require a change to GraphData, so I opted out of that. 

The other question is whether we want to enable limits on the RMSD error for a conformer to be returned, because if you set self.num_conformers too high you will get unrealistic molecules. Not sure how important that is, or if we can leave that up to the user to decide.

File: deepchem/feat/molecule_featurizers/conformer_featurizer.py [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    >>> import numpy as np
+    >>> featurizer = RDKitConformerFeaturizer(num_conformers=2)
+    >>> molecule = "CCO"
+    >>> features_list = featurizer.featurize([molecule])
+    >>> batched_feats = BatchGraphData(np.concatenate(features_list).ravel())
+    >>> print(batched_feats.node_pos_features.shape)
+    (18, 3)
+    """
+
+    def __init__(self, num_conformers: int = 1):
```
Comment (rbharath) [inline deepchem/feat/molecule_featurizers/conformer_featurizer.py:105]: Can you a docstring?

**Review by rbharath — COMMENTED** [review]
Documenting offline discussion, @tonydavis629 to proceed with fixed number of conformers, a quality rejection threshold for conformers, and to either round-robin repeat or use Nones in the case that there are insufficient high quality conformers.

File: deepchem/feat/molecule_featurizers/conformer_featurizer.py (hunk 1 above)
Comment (tonydavis629) [reply deepchem/feat/molecule_featurizers/conformer_featurizer.py:7]: done

File: deepchem/feat/molecule_featurizers/conformer_featurizer.py (hunk 5 above)
Comment (tonydavis629) [reply deepchem/feat/molecule_featurizers/conformer_featurizer.py:105]: done

**Pull request merged**